src/BenchmarkDotNet.TestAdapter/VSTestEventProcessor.cs

```diff
@@ -0,0 +1,187 @@
+﻿using BenchmarkDotNet.EventProcessors;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.TestAdapter.Remoting;
+using BenchmarkDotNet.Toolchains.Results;
+using BenchmarkDotNet.Validators;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+
+namespace BenchmarkDotNet.TestAdapter
+{
+    /// <summary>
+    /// An event processor which will pass on benchmark execution information to VSTest.
+    /// </summary>
+    internal class VSTestEventProcessor : EventProcessor
```

Comment:
I would suggest using `Vs` instead of `VS` in all symbol names. Motivation:
* Miscorosft tend to use `Vs` in their own projects, e.g., see https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/VsTestConsoleProcessManager.cs
* Names `VSTestEventProcessor` cause warnings for some linters. `Vs` is a more safe way to avoid warnings with different tools.